src/aegis/src/aegis.erl

```diff
@@ -36,58 +33,40 @@ create(#{} = Db, Options) ->
         db_prefix := DbPrefix
     } = Db,
 
-    % Generate new key
-    DbKey = crypto:strong_rand_bytes(32),
+    {ok, AegisConfig} = aegis_server:generate_key(Db, Options),
 
-    % protect it with root key
-    WrappedKey = aegis_keywrap:key_wrap(?ROOT_KEY, DbKey),
-
-    % And store it
     FDBKey = erlfdb_tuple:pack(?WRAPPED_KEY, DbPrefix),
-    ok = erlfdb:set(Tx, FDBKey, WrappedKey),
+    Packed = erlfdb_tuple:pack({AegisConfig}),
+    ok = erlfdb:set(Tx, FDBKey, Packed),
 
     Db#{
-        aegis => DbKey
+        aegis => AegisConfig
     }.
 
 
-open(#{} = Db, Options) ->
+open(#{} = Db, _Options) ->
     #{
         tx := Tx,
         db_prefix := DbPrefix
     } = Db,
 
-    % Fetch wrapped key
     FDBKey = erlfdb_tuple:pack(?WRAPPED_KEY, DbPrefix),
-    WrappedKey = erlfdb:wait(erlfdb:get(Tx, FDBKey)),
-
-    % Unwrap it
-    DbKey = aegis_keywrap:key_unwrap(?ROOT_KEY, WrappedKey),
+    Packed = erlfdb:wait(erlfdb:get(Tx, FDBKey)),
+    {AegisConfig} = erlfdb_tuple:unpack(Packed),
```

Comment:
Perhaps it's the job of the key manager to do this? Not all key managers might need to read or write to erlfdb, so perhaps they should do their own persistence, and just return a value that aegis.erl puts in the the Db meta map, but does not save it or transform it at all.

Reply:
e.g,

```
{ok, AegisConfig} = aegis_server:open_db(Db, Options),
Db#{
    aegis = AegisConfig
}
```

in aegis_server, we call the aegis key manager with the `Db` and `Options` and our one will read the appropriate value from fdb (containing the wrapped key).

At the moment we have a mixture of aegis and the key manager jointly responsible for a piece of data. The key manager isn't free to return what it likes, because aegis will try to pack it into a tuple, which will fail for many inputs.

Reply:
Very good point, let me try to do this to confirm there are no hidden constrains like breaking of original process boundary on erlfdb's transaction.

What do you think if we make `key_manager` to serialize `AegisConfig` value it's getting from `?AEGIS_KEY_MANAGER` callback with `term_to_binary`, so from the callback's perspective there will be no limitations on `AegisConfig` format?